### PR TITLE
[9.0] [MIG] purchase_order_type module

### PR DESCRIPTION
**Pull request by cubells**
- Updated README
- Updated views
- Removed invoice_method
- Updated onchange_partner_id method

cc @Tecnativa

### CODE REVIEW

File: purchase_order_type/README.rst [hunk 1]
```diff
@@ -1,6 +1,8 @@
 .. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
-    :alt: License: AGPL-3
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
 
+===================
 Purchase Order Type
 ===================
```
Comment (pedrobaeza) [inline purchase_order_type/README.rst]: Include in the configuration section that you can predefine a purchase order type in the partner.

Include in the usage section that the type will be autofilled when selecting a partner that has a type set in its data.

File: purchase_order_type/models/purchase_order.py [hunk 2]
```diff
@@ -36,19 +20,10 @@ def _default_order_type(self):
...
-                    'order_type': partner.purchase_type.id,
-                })
-        return res
-
-    @api.one
-    @api.onchange('order_type')
-    def onchange_order_type(self):
-        self.invoice_method = self.order_type.invoice_method
+    @api.onchange('partner_id', 'company_id')
+    def onchange_partner_id(self):
```
Comment (pedrobaeza) [inline purchase_order_type/models/purchase_order.py]: If you don't want to call super as this is an isolate operation, you can do:
```
    @api.onchange('partner_id', 'company_id')
    def onchange_partner_id_purchase_order_type(self):
        if self.partner_id.purchase_type:
            self.order_type = self.partner_id.purchase_type.id
```
Although you don't do this change, change the `if` part to the new API possibilities (no need to check the existence of each field before accessing the subfields).

File: purchase_order_type/models/purchase_order_type.py [hunk 3]
```diff
@@ -1,42 +1,13 @@
...
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-#
+# Copyright 2015 Guewen Baconnier <guewen.baconnier@camptocamp.com>
+# Copyright 2016 Vicent Cubells <vicent.cubells@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
-from openerp import models, fields, api
+from openerp import models, fields
```
Comment (pedrobaeza) [inline purchase_order_type/models/purchase_order_type.py]: Order alphabetically

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
For giving any functionality added, you can add the stock picking type (only incoming ones), and incoterm information to purchase order types, and pass this information when changing the order type.

**Comment by pedrobaeza** [comment]
What about the last point?:
```
For giving any functionality added, you can add the stock picking type (only incoming ones), and incoterm information to purchase order types, and pass this information when changing the order type.
```

File: purchase_order_type/models/purchase_order.py [hunk 4]
```diff
@@ -24,3 +24,9 @@ def _default_order_type(self):
     def onchange_partner_id_purchase_order_type(self):
         if self.partner_id.purchase_type:
             self.order_type = self.partner_id.purchase_type.id
+
+    @api.onchange('order_type')
+    def onchange_purchase_order_type(self):
+        if self.order_type:
+            self.incoterm_id = self.order_type.incoterm_id.id
```
Comment (pedrobaeza) [inline purchase_order_type/models/purchase_order.py]: This should be both also conditional. If the value is set, then you overwrite whatever it has. If not, you leave the previous value.

**Review by pedrobaeza — APPROVED** [review]
You can add also instructions for the 2 new fields, but not blocking. 9.0 commits to be squashed on merge.

**Review by jjscarafia — APPROVED** [review]

**Pull request merged**